flaky test TestDoImport

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```
=== RUN   TestDoImport
    local_test.go:2086: 
        	Error Trace:	pkg/lightning/backend/local/local_test.go:2086
        	Error:      	"[{{0xc001994000 <nil>} { <nil>}} {{<nil> <nil>} {ingested <nil>}}]" should have 0 item(s), but has 2
        	Test:       	TestDoImport
--- FAIL: TestDoImport (1.00s)
```
after we add load balancer, the order of jobs depends on map iteration order, and `fatal error` job might be run first
https://github.com/pingcap/tidb/blob/4df3389c263d5acd1c19155f280eb015b757cd76/pkg/lightning/backend/local/region_job.go#L1170-L1172

add a sleep before `runSendToWorker` will reproduce it easily
https://github.com/pingcap/tidb/blob/4df3389c263d5acd1c19155f280eb015b757cd76/pkg/lightning/backend/local/region_job.go#L1088-L1089

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)
master
<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/lightning/backend/local/local_test.go
+++ b/pkg/lightning/backend/local/local_test.go
@@ -2081,11 +2081,6 @@ func TestDoImport(t *testing.T) {
 	}
 	err = l.doImport(ctx, e, initRegionKeys, int64(config.SplitRegionSize), int64(config.SplitRegionKeys))
 	require.ErrorContains(t, err, "fatal error")
-	for _, v := range fakeRegionJobs {
-		for _, job := range v.jobs {
-			require.Len(t, job.injected, 0)
-		}
-	}
 }
 
 func TestRegionJobResetRetryCounter(t *testing.T) {

